src/OfflineWorkflowsSample/OfflineWorkflowsSample/ViewModels/ItemPages/Offline/GenerateMapAreaViewModel.cs

```diff
@@ -0,0 +1,352 @@
+            }
+        }
+
+        public int MaximumLevelOfDetail
+        {
+            get => _maximumLevelOfDetail;
+            set => SetProperty(ref _maximumLevelOfDetail, value);
+        }
+
+        public string SelectedMaximumScale
+        {
+            get => _selectedMaximumScale;
+            set => SetProperty(ref _selectedMaximumScale, value);
+        }
+
+        public string SelectedMinimumScale
+        {
+            get => _selectedMinimumScale;
+            set => SetProperty(ref _selectedMinimumScale, value);
+        }
+
+        public int SelectedLevelOfDetail
+        {
+            get => _selectedLevelOfDetail;
+            set
+            {
+                SetProperty(ref _selectedLevelOfDetail, value);
+                UpdateScales();
+            }
+        }
+
+        public bool IsMapOnline => _map.Item is PortalItem;
+
+        #endregion Properties
+
+        #region Misc. Overhead
+
+        private IWindowService _windowService;
+        private MapViewService _mapViewService;
+
+        public GenerateMapAreaViewModel()
+        {
+            _generateMapAreaCommand = new DelegateCommand(GenerateMapArea, () => IsMapOnline);
+            _navigateToMapAreaCommand = new DelegateCommand(NavigateToMapArea, () => IsMapOnline);
+            _openMapFileCommand = new DelegateCommand(RevealInExplorer, () => !IsMapOnline);
+            _resetMapCommand = new DelegateCommand(() => RaiseMapChanged(true), () => !IsMapOnline);
+        }
+
+        public async Task Initialize(Map map, IWindowService windowService, MapViewService mapViewService)
+        {
+            _windowService = windowService;
+            _mapViewService = mapViewService;
+            Map = map;
+            try
+            {
+                _windowService.SetBusyMessage("Loading map...");
+                _windowService.SetBusy(true);
+
+                await Map.LoadAsync();
+
+                try
+                {
+                    // Set the scale information to the UI
+                    var basemapLayer = Map.AllLayers.OfType<ImageTiledLayer>().FirstOrDefault();
+                    if (basemapLayer != null)
+                    {
+                        _levelsOfDetail = basemapLayer.TileInfo.LevelsOfDetail;
+                        MaximumLevelOfDetail = _levelsOfDetail.Max(x => x.Level);
+                    }
+                    else if (Map.AllLayers.OfType<ArcGISVectorTiledLayer>().Any())
+                    {
+                        var vectorBasemapLayer = Map.AllLayers.OfType<ArcGISVectorTiledLayer>().First();
+                        _levelsOfDetail = vectorBasemapLayer.SourceInfo.LevelsOfDetail;
+                        MaximumLevelOfDetail = _levelsOfDetail.Max(x => x.Level);
+                    }
+                }
+                catch (Exception)
+                {
+                    _levelsOfDetail = new List<LevelOfDetail>();
+                }
+
+                SelectedLevelOfDetail = 17;
```

Comment:
There could be fewer than 18 levels-of-detail available. It'd be best to calculate this number based on the size of `_levelsOfDetail`.